src/App.tsx

```diff
@@ -15,48 +15,121 @@ export const goodsFromServer = [
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort((a, b) => a.localeCompare(b));
+      break;
+
+    case SortType.LENGTH:
+      visibleGoods.sort((a, b) => a.length - b.length);
+      break;
+
+    default:
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortGoodsByAlphabet = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const sortGoodsByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const reverseGoods = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const resetGoods = () => {
+    setSortType(SortType.NONE);
+    setIsReversed(false);
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={
+            sortType === SortType.ALPHABET
+              ? 'button is-info'
+              : 'button is-info is-light'
+          }
+          onClick={sortGoodsByAlphabet}
         >
           Sort alphabetically
-        </button>
 
+        </button>
         <button
           type="button"
-          className="button is-success is-light"
+          className={
+            sortType === SortType.LENGTH
+              ? 'button is-info'
+              : 'button is-info is-light'
+          }
+          onClick={sortGoodsByLength}
         >
           Sort by length
-        </button>
 
+        </button>
         <button
           type="button"
-          className="button is-warning is-light"
+          className={
+            isReversed === true
+              ? 'button is-info'
+              : 'button is-info is-light'
+          }
+          onClick={reverseGoods}
         >
           Reverse
-        </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
         </button>
+        {(sortType || isReversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={resetGoods}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {getReorderedGoods(goodsFromServer, { sortType, isReversed }).map(
```

Comment:
i think you were previously asked to move `getReorderedGoods` to separate const/var

Reply:
![image](https://github.com/mate-academy/react_list-of-goods-hooks/assets/55200530/402f9c37-8443-4ec7-a98d-2237de1f98c1)
 
 
 move it to variable